server/repository/src/migrations/v2_00_00/sync_file_reference.rs

```diff
@@ -1,6 +1,42 @@
 use crate::migrations::*;
 
 pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    #[cfg(feature = "postgres")]
+    sql!(
+        connection,
+        r#"
+        CREATE TYPE sync_file_status AS ENUM (
+            'NEW',
+            'IN_PROGRESS',
+            'ERROR',
+            'DONE',
+            'PERMANENT_FAILURE'
+        );
+        CREATE TYPE sync_file_direction AS ENUM (
+            'UPLOAD',
+            'DOWNLOAD'
+        );
+        CREATE TABLE sync_file_reference (
+            id TEXT NOT NULL PRIMARY KEY,
+            table_name TEXT NOT NULL, -- Associated Table
+            record_id TEXT NOT NULL, -- Associated record id
+            file_name TEXT NOT NULL,
+            mime_type TEXT,
+            uploaded_bytes INTEGER NOT NULL DEFAULT 0,
+            downloaded_bytes INTEGER NOT NULL DEFAULT 0,
+            total_bytes INTEGER NOT NULL DEFAULT 0,
+            retries INTEGER NOT NULL DEFAULT 0,
+            retry_at TIMESTAMP,
+            direction sync_file_direction NOT NULL,
+            status sync_file_status NOT NULL,
+            error TEXT,
+            created_datetime TIMESTAMP NOT NULL, -- No modified datetime, as we don't allow updates it would break sync
+            deleted_datetime TIMESTAMP
+        );
+        "#,
+    )?;
+
+    #[cfg(not(feature = "postgres"))]
```

Comment:
I don't like the approach here where we have 2 table definitions, which could more easily be miss matched between sqlite and postgres, Mark has this approach which I prefer...


Define the type as a const
``` 
   const DIRECTION_ENUM_TYPE: &str = if cfg!(feature = "postgres") {
        "sync_file_direction"
    } else {
        "TEXT"
    };
```

Create the enums for postgres
```
    #[cfg(feature = "postgres")]
    sql!(
        connection,
        r#"
        CREATE TYPE sync_file_status AS ENUM (
            'NEW',
            'IN_PROGRESS',
            'ERROR',
            'DONE',
            'PERMANENT_FAILURE'
        );
        CREATE TYPE sync_file_direction AS ENUM (
            'UPLOAD',
            'DOWNLOAD'
        );#,
);
```

Then we can have a single table definition like this:
```
    sql!(
        connection,
        r#"
            CREATE TABLE sync_file_reference (
                id TEXT NOT NULL PRIMARY KEY,
                table_name TEXT NOT NULL, -- Associated Table
                record_id TEXT NOT NULL, -- Associated record id
                file_name TEXT NOT NULL,
                mime_type TEXT,
                uploaded_bytes INTEGER NOT NULL DEFAULT 0,
                downloaded_bytes INTEGER NOT NULL DEFAULT 0,
                total_bytes INTEGER NOT NULL DEFAULT 0,
                retries INTEGER NOT NULL DEFAULT 0,
                retry_at TIMESTAMP,
                direction {DIRECTION_ENUM_TYPE} NOT NULL,
                status TEXT NOT NULL,
                error TEXT,
                created_datetime TIMESTAMP NOT NULL, -- No modified datetime, as we don't allow updates it would break sync
                deleted_datetime TIMESTAMP
            );
        "#,
    )?;
```

Similar for status etc

Reply:
Done now. 

```
git difftool -d --tool=kdiff3 update-diesel:server/repository/src/migrations..develop:server/repository/src/migrations
```

Shows now changes to 2.0.0 migrations now